### DISCUSSION TITLE
Access all DB Connections via 'MikroORMs' Injection token

### DISCUSSION THREAD

**Discussion by rogrdat** [mikro-orm/nestjs#87] [Q&A]
Consider an application where -- 
- Users are queries from a generic 'System' DB/Cache
- Transactional (context-specifc) application data is a - 'Context' DB (RDBMS)
- Queries are execute in (context-specific) but distributed/de-normalized DB (Ex: Mongo)

Based on the User/Context (tenant/geo) /Action I might have to switch the specific ORM connection to use at Runtime.  Based on reviews, an approach would to be try to access the 'MikroOrms' array via Injection.  So all such as 
```async myAction(@Inject('MikroOrms') private readonly orms: MikroORM<PostgresqlDriver>[])```  
should grant me in the application code full access to all the DB connections allowing me to determine the relevant one to use.  

But the following code errors out -- the error I see is this --
ERROR: Nest can't resolve dependencies of the CompanyService (I18nService, ?). Please make sure that the argument MikroORMs at index [1] is available in the CompanyModule context.  

@tsangste and others.., any Ideas or thoughts?

Here is my code to register the OrmModule
```
@Global()
@Module({
	imports: [
		MikroOrmModule.forRootAsync({
			imports: [ConfigModule],
			contextName: 'system',
			useFactory: (configService: ConfigService) => ({
				type: 'postgresql',
				host: configService.get('database.host'),
				port: configService.get<number>('database.port'),
				password: configService.get('database.password'),
				user: configService.get('database.username'),
				dbName: configService.get('database.dbName'),
				entities: ['dist/entities/**/*.entity.js'],
				entitiesTs: ['src/entities/**/*.entity.ts'],
				debug: true,
				loadStrategy: LoadStrategy.JOINED,
				highlighter: new SqlHighlighter(),
				metadataProvider: TsMorphMetadataProvider,
				entityRepository: BaseRepository,
				allowGlobalContext: false,
				registerRequestContext: false,
				pool: { min: 2, max: 10 },
			}),
			inject: [ConfigService],
		}),
		MikroOrmModule.forMiddleware(),
		MikroOrmModule.forFeature({
			entities: BASE_ENTITIES,
			contextName: 'system',
		}),
	],
	exports: [MikroOrmModule],
})
export class OrmModule {}
```
I am trying to access the `MikroORMs` token as described[ here](https://github.com/mikro-orm/nestjs/blob/f8c68773afd6989dfeb27dae7ce63a1946e91099/src/multiple-mikro-orm.middleware.ts#L9)
```
@Injectable()
export class CompanyService {
	constructor(private readonly i18nService: I18nService, @Inject('MikroORMs') private readonly orms: MikroORM[]) {}
}
```
The errors reported are -- 
```
_[Nest] 13323  -   ERROR [ExceptionHandler] Nest can't resolve dependencies of the CompanyService (I18nService, ?). Please make sure that the argument MikroORMs at index [1] is available in the CompanyModule context.

Potential solutions:
- If MikroORMs is a provider, is it part of the current CompanyModule?
- If MikroORMs is exported from a separate @Module, is that module imported within CompanyModule?
  @Module({
    imports: [ /* the Module containing MikroORMs */ ]
  })_
```

**Comment by tsangste** [comment]
The changes I made for multiple database connections with the `MikroORMs` token isn't exported and therefore is not consumable outside of the module.

You can create a PR to make it available for use outside of that by exporting the token.

**Reply by rogrdat** [reply]
PR Submitted.  Have tested to confirm that exporting the token 'MikroORMs' opens up access to the connections.  While this addresses my needs, am concerned about best practices around connection management, etc.
